### PR TITLE
[REF] crm_project_issue:  new api

### PR DESCRIPTION
**Pull request by hmo-odoo**
Task: https://www.odoo.com/web#id=12140&view_type=form&model=project.task&menu_id=3942&action=327

Pad: https://pad.odoo.com/p/r.FvY4KjvA7Amr2gKM

### CODE REVIEW

File: addons/crm_project_issue/wizard/crm_lead2projectissue_wizard.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp import api, fields, models, _
```
Comment (atp-odoo) [inline addons/crm_project_issue/wizard/crm_lead2projectissue_wizard.py]: use from odoo import api, fields, models

File: addons/crm_project_issue/views/crm_lead_views.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+<odoo>
+
+    <!-- inherit view of Lead Form : adding the "convert to issue" button -->
+    <record id="crm_case_form_view_leads_project_issue" model="ir.ui.view">
+        <field name="name">crm.lead.form.inherit.lead2projectissue</field>
+        <field name="model">crm.lead</field>
+        <field name="inherit_id" ref="crm.crm_case_form_view_leads" />
+        <field name="arch" type="xml">
+            <xpath expr="//div[@name='button_box']" position="inside">
+                    <button class="oe_inline oe_stat_button" type="action" name="%(convert_lead2projectissue_wizard_action)d" icon="fa-bug" help="Convert to Issue">
```
Comment (atp-odoo) [inline addons/crm_project_issue/views/crm_lead_views.xml]: remove this class oe_inline. no need to this css class.

File: addons/crm_project_issue/__openerp__.py [hunk 3]
```diff
@@ -12,7 +14,8 @@
 Link module to map leads to issues
         """,
     'data': [
-        'project_issue_view.xml'
+        'wizard/crm_lead2projectissue_wizard_view.xml',
+        'views/crm_lead_views.xml'
     ],
     'depends': ['crm', 'project_issue'],
     'installable': True,
```
Comment (atp-odoo) [inline addons/crm_project_issue/__openerp__.py]: remove some attribute which contain default values: 
version, installable,

File: addons/crm_project_issue/wizard/crm_lead2projectissue_wizard.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+from odoo import api, fields, models, _
+
+
+class CrmLeadToProjectIssueWizard(models.TransientModel):
+    """ wizard to convert a Lead into a Project Issue and move the Mail Thread """
+    _name = "crm.lead2projectissue.wizard"
+    _inherit = 'crm.partner.binding'
+
+    lead_id = fields.Many2one('crm.lead', string='Lead', domain=[('type', '=', 'lead')], default=lambda self: self.env.context.get('active_id'))
```
Comment (atp-odoo) [inline addons/crm_project_issue/wizard/crm_lead2projectissue_wizard.py]: is this field required ? cannot we remove this field ? we can directly get lead_id from context with active_id of lead.even not displayed in view side..used here to just take value from lead which we directly get from context.get('active_id').

File: addons/crm_project_issue/wizard/crm_lead2projectissue_wizard.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    _inherit = 'crm.partner.binding'
+
+    project_id = fields.Many2one('project.project', string='Project', domain=[('use_issues', '=', True)])
+
+    @api.multi
+    def action_lead_to_project_issue(self):
+        # get the models
+        Issue = self.env['project.issue']
+        for wizard in self:
+            self.model = self.env.context.get('active_model')
```
Comment (atp-odoo) [inline addons/crm_project_issue/wizard/crm_lead2projectissue_wizard.py]: make condition this way : 
if (self.env.context.get('active_model') == 'crm.lead') and self.env.context.get('active_id'):
            active_model = self.env['crm.lead'].browse(self.env.context.get('active_id'))

File: addons/crm_project_issue/wizard/crm_lead2projectissue_wizard.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    _inherit = 'crm.partner.binding'
+
+    project_id = fields.Many2one('project.project', string='Project', domain=[('use_issues', '=', True)])
+
+    @api.multi
+    def action_lead_to_project_issue(self):
+        # get the models
+        Issue = self.env['project.issue']
+        for wizard in self:
+            lead = self.env['crm.lead'].browse(self.env.context.get('active_id'))
```
Comment (hmo-odoo) [inline addons/crm_project_issue/wizard/crm_lead2projectissue_wizard.py]: can browse lead record before for loop ?

File: addons/crm_project_issue/__openerp__.py [hunk 7]
```diff
@@ -12,8 +13,8 @@
 Link module to map leads to issues
         """,
     'data': [
-        'project_issue_view.xml'
+        'wizard/crm_lead2projectissue_wizard_view.xml',
+        'views/crm_lead_views.xml'
```
Comment (WHPODOO) [inline addons/crm_project_issue/__openerp__.py]: please arrange alphabetically like..
```
A for apple
B for Ball or Bird
```
:laughing: 
alphabet-coloring-pages-printable-apple-and-bird

File: addons/crm_project_issue/views/crm_lead_views.xml [hunk 8]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+
+    <!-- inherit view of Lead Form : adding the "convert to issue" button -->
```
Comment (WHPODOO) [inline addons/crm_project_issue/views/crm_lead_views.xml]: @rpi-odoo  can you please tell me what is the need of this blank line ??
its nothing just increase LOC.

File: addons/crm_project_issue/views/crm_lead_views.xml [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+                        <div class="o_form_field o_stat_info">
+                            <span class="o_stat_text">Convert To</span>
+                            <span class="o_stat_text">Issue</span>
+                        </div>
+                    </button>
+            </xpath>
+        </field>
+    </record>
+
+</odoo>
```
Comment (WHPODOO) [inline addons/crm_project_issue/views/crm_lead_views.xml]: @rpi-odoo  can you please tell me what is the need of this blank line ??
its nothing just increase LOC.

File: addons/crm_project_issue/wizard/crm_lead2projectissue_wizard.py [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+            # create new project.issue
+            vals = {
+                "name": lead.name,
+                "description": lead.description,
+                "email_from": lead.email_from,
+                "project_id": wizard.project_id.id,
+                "partner_id": partner_id,
+                "user_id": None
+            }
+            issue = Issue.create(vals)
```
Comment (WHPODOO) [inline addons/crm_project_issue/wizard/crm_lead2projectissue_wizard.py]: Hey @rpi-odoo you are spending unusually bytes behind variables so please keep pin it.
When you use once then assign directly. like : 
```
issue = self.env['project.issue'].create(vals)
```

File: addons/crm_project_issue/wizard/crm_lead2projectissue_wizard.py [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+    """ wizard to convert a Lead into a Project Issue and move the Mail Thread """
+    _name = "crm.lead2projectissue.wizard"
+    _inherit = 'crm.partner.binding'
+
+    project_id = fields.Many2one('project.project', string='Project', domain=[('use_issues', '=', True)])
+
+    @api.multi
+    def action_lead_to_project_issue(self):
+        # get the models
+        Issue = self.env['project.issue']
```
Comment (WHPODOO) [inline addons/crm_project_issue/wizard/crm_lead2projectissue_wizard.py]: Remove this object reference variable and check below comment.
look-below

File: addons/crm_project_issue/__openerp__.py (hunk 7 above)
Comment (rpi-odoo) [inline addons/crm_project_issue/__openerp__.py]: @WHPODOO This time you are wrong. Just go through crm_lead_views.xml file you can get convert_lead2projectissue_wizard_action called there and that record is in wizard file. So the wizard file should be placed first here. 
636x460design_01

Think twice before commenting on any PR.
And one more thing I have learned A,B,C,D in my Nursery.  :D

Comment (WHPODOO) [inline addons/crm_project_issue/__openerp__.py]: @rpi-odoo 
Hey man Oops :laughing: ,
I will defiantly think,
Its my great mistake,
But you really needs revise your code.
Then when you free take change of Nursery , Actually you need it.
And as per your code take only your coffee that's good.

ping-pong-tongue-animated-frogs-breakfast

**Pull request merged**